File: week2-math/problem09-N개의최소공배수/heejae.cpp
# 문제: N개의 최소공배수
# 접근 방식: 먼저 배열의 첫번째, 두번째 원소끼리의 lcm을 구한다. -> 배열의 세번째 원소부터 기존 lcm과의 lcm을 다시 구한다.
# 시간복잡도: O(n × log M)

#include <string>
#include <vector>
#include <cmath>
#include <numeric>

using namespace std;

int solution(vector<int> arr) {
    int answer = std::lcm(arr[0], arr[1]);
        
    for(int i=2; i<arr.size(); i++) {
        answer = std::lcm(answer, arr[i]);
    }
        
    return answer;
}

File: week1-array-string/problem05-같은숫자는싫어/heejae.java
// 문제: 같은 숫자는 싫어
// 접근 방식: 현재 넣으려는 숫자(arr[i])가 list에 없을 때만 넣는다.
// 시간복잡도: O(n)

import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        
        List<Integer> list = new ArrayList<>();
        
        for(int i=0; i<arr.length; i++) {
            if(list.size() > 0 && list.get(list.size()-1) == arr[i]) continue;
            
            list.add(arr[i]);
        }
        
        int[] answer = new int[list.size()];
        for(int i=0; i<list.size(); i++) answer[i] = list.get(i);
        
        return answer;
    }
}
